rskj-core/src/test/java/co/rsk/peg/RegisterBtcTransactionIT.java

```diff
@@ -0,0 +1,217 @@
+package co.rsk.peg;
+
+import co.rsk.bitcoinj.core.*;
+import co.rsk.bitcoinj.script.ScriptBuilder;
+import co.rsk.core.RskAddress;
+import co.rsk.crypto.Keccak256;
+import co.rsk.bitcoinj.store.BlockStoreException;
+import co.rsk.peg.bitcoin.BitcoinTestUtils;
+import co.rsk.peg.btcLockSender.BtcLockSenderProvider;
+import co.rsk.peg.constants.BridgeConstants;
+import co.rsk.peg.constants.BridgeMainNetConstants;
+import co.rsk.peg.federation.*;
+import co.rsk.peg.feeperkb.FeePerKbSupport;
+import co.rsk.peg.storage.BridgeStorageAccessorImpl;
+import co.rsk.peg.storage.StorageAccessor;
+import co.rsk.test.builders.BridgeSupportBuilder;
+import co.rsk.test.builders.FederationSupportBuilder;
+import org.bouncycastle.util.encoders.Hex;
+import org.ethereum.config.blockchain.upgrades.ActivationConfig;
+import org.ethereum.config.blockchain.upgrades.ActivationConfigsForTest;
+import org.ethereum.core.*;
+import org.ethereum.vm.PrecompiledContracts;
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.*;
+
+import java.io.IOException;
+import java.math.BigInteger;
+import java.util.*;
+
+import static org.mockito.Mockito.*;
+
+public class RegisterBtcTransactionIT {
+
+    private static final BridgeConstants bridgeConstants = BridgeMainNetConstants.getInstance();
+    private static final NetworkParameters btcParams = bridgeConstants.getBtcParams();
+    private final BridgeSupportBuilder bridgeSupportBuilder = BridgeSupportBuilder.builder();
+
+    @Test
+    void whenRegisterALegacyBtcTransactionTheBridgeShouldRegisterTheNewUtxoAndTransferTheRbtcBalance() throws BlockStoreException, AddressFormatException, IOException, BridgeIllegalArgumentException {
+        ActivationConfig.ForBlock activationConfig = ActivationConfigsForTest.all().forBlock(0);
+        Repository repository = BridgeSupportTestUtil.createRepository();
+        Repository track = repository.startTracking();
+        Block rskExecutionBlock = getRskExecutionBlock();
+
+        BtcLockSenderProvider btcLockSenderProvider = new BtcLockSenderProvider();
+        FeePerKbSupport feePerKbSupport = getFeePerKbSupport();
+
+        Federation federation = P2shErpFederationBuilder.builder().build();
+        FederationStorageProvider federationStorageProvider = getFederationStorageProvider(track, federation);
+        FederationSupport federationSupport = getFederationSupport(federationStorageProvider, activationConfig);
```

Comment:
I notice there are a some methods that build these adapter variables. Make sure of declaring them as instance variables and building them before running the each test since you will be reusing them. Check this [example](https://github.com/rsksmart/rskj/pull/2874/files#diff-254b94dfb4a05ec990094a9d5fd3b2ebae191a2e6d9047f897341ce3216e42c2R17-R20).

Reply:
Since there is only one test I left the setUp in the test, for future tests I extract them to the setUp